packages/hub/src/hooks/useAsyncMutation.ts

```diff
@@ -21,6 +21,22 @@ export type CommonMutationParameters<TTypename extends string> =
     };
   };
 
+type UseAsyncMutationOkResult<
+  TMutation extends CommonMutationParameters<TTypename>,
+  TTypename extends string = string,
+> = Extract<TMutation["response"]["result"], { __typename: TTypename }>;
+
+export type UseAsyncMutationAct<
```

Comment:
Exposing this type allowed me to extract `onCompleted` type in `MutationButton`.